keiko-dynomite/src/main/kotlin/com/netflix/spinnaker/q/dynomite/DynomiteQueue.kt

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2017 Netflix, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.netflix.spinnaker.q.dynomite
+
+import com.fasterxml.jackson.databind.ObjectMapper
+import com.google.common.hash.Hashing
+import com.netflix.dyno.queues.DynoQueue
+import com.netflix.dyno.queues.redis.RedisQueues
+import com.netflix.spinnaker.q.AttemptsAttribute
+import com.netflix.spinnaker.q.MaxAttemptsAttribute
+import com.netflix.spinnaker.q.Message
+import com.netflix.spinnaker.q.Queue
+import com.netflix.spinnaker.q.QueueCallback
+import com.netflix.spinnaker.q.metrics.EventPublisher
+import com.netflix.spinnaker.q.metrics.MessageDead
+import com.netflix.spinnaker.q.metrics.MonitorableQueue
+import com.netflix.spinnaker.q.metrics.QueueState
+import com.netflix.spinnaker.q.metrics.fire
+import org.slf4j.LoggerFactory
+import java.nio.charset.Charset
+import java.time.Duration
+import java.time.temporal.ChronoUnit
+import java.time.temporal.TemporalAmount
+import java.util.concurrent.TimeUnit
+
+// TODO rz - Currently only supports a single dyno queue shard
+class DynomiteQueue(
+  private val redisQueues: RedisQueues,
+  private val mapper: ObjectMapper,
+  override val ackTimeout: TemporalAmount = Duration.ofMinutes(1),
+  override val deadMessageHandler: (Queue, Message) -> Unit,
+  override val publisher: EventPublisher
+) : MonitorableQueue {
+
+  private val log = LoggerFactory.getLogger(javaClass)
+
+  // The queue implementation exposes a decent amount of metrics own its own.
+  override fun readState() = QueueState(
+      depth = queue().size().toInt(),
+      ready = 0,
+      unacked = 0,
+      orphaned = 0,
+      hashDrift = 0
+    )
+
+  override fun poll(callback: QueueCallback) {
+    queue().pop(1, 1, TimeUnit.SECONDS)
+      .firstOrNull()
+      ?.also { message ->
```

Comment:
I don't have any error handling code on poll yet. This is the biggest functional difference between Redis/Dynomite backends at the moment.